src/main/flight/mixer.c

```diff
@@ -710,6 +710,25 @@ static void applyMixToMotors(float motorMix[MAX_SUPPORTED_MOTORS])
     }
 }
 
+float applyThrottleLimit(float throttle)
+{
+    if (currentControlRateProfile->throttle_limit_percent < 100) {
+        const float throttleLimitFactor = currentControlRateProfile->throttle_limit_percent / 100.0f;
+        switch (currentControlRateProfile->throttle_limit_type) {
+            case THROTTLE_LIMIT_TYPE_SCALE:
+                throttle = throttle * throttleLimitFactor;
```

Comment:
Since it's a separate function now, I would replace this with:
`return throttle * throttleLimitFactor`